sdk/go/common/apitype/core.go

```diff
@@ -329,6 +329,10 @@ type ResourceV3 struct {
 	// If set, the providers Delete method will not be called for this resource
 	// if specified resource is being deleted as well.
 	DeletedWith resource.URN `json:"deletedWith,omitempty" yaml:"deletedWith,omitempty"`
+	// Created tracks when this resource was created if known. Checkpoints prior to early 2023 do not include this.
```

Comment:
Agreed that was my thinking, should have been more clear. I just want those names defined now we don't need to actually plumb it all through for this PR.

Reply:
> I think to help with names we should add the other obvious time fields here _now_.
> 
> 1. When Create was called
> 2. When the state was first written (via create or import)
> 3. When the state was last updated in any way (create/update/import/refresh)
> 4. When the resource was last updated (via create or update)

I think it's worth postponing even leaving comments suggesting design on this until we have a problem meriting a design because the ideas we have now may not be the best solution for the problem we aren't aware of yet.

The discussions around semantics have been captured in our PR discussion and I'm sure we will remember that there was discussion and can reference this PR when discussing this in the future.

Note to self: see internal poll for field naming.

Reply:
I talked to @blampe and he says that the service may only use this for ordering resources based on most recently updated to make it easier for users to identify issues.

I think we don't have to worry about this growing and can just merge it.

Reply:
If that's the case we should remove CreatedAt

Reply:
On the Service side we would love a CreatedAt field. What Bryce is saying is why we want last updated, which is the default sort in the console will be to last updated descending. But these fields, CreatedAt included, are needed for a much larger use case: analytics. We are providing customers the ability to export this data to CSV and then build their own reports, dashboards, queries, etc. from it. These fields will make these reports much more rich.

Let me know what else is needed to get this across the line.

Reply:
So if we have CreatedAt which records when Create is called isn't it highly likely we'll want XyzAt which records when Update is called?

Reply:
Although if we're changing to CreatedAt and ModifiedAt that at least leaves the obvious UpdatedAt for actually calling update. Still not super keen on how there's nothing in the names to make it clear these are talking about different entities.